### PR TITLE
tracking user profile views from the extension

### PR DESCRIPTION
**Pull request by 2is10**
@yipingliao

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/app.js [hunk 1]
```diff
@@ -69,49 +69,35 @@ angular.module('kifi', [
...
+      $location.search(_.omit(search, params)).replace();
     }
-
-    return {
-      state: state,
-      pushState: pushState
-    };
+    return state;
   }
 ])
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/app.js:83]: @andrewconner I hope this change to `injectedState` is okay with you. It looks like `pushState` wasn’t being used.

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/app.js:83]: Note that we now strip the specific params that we’re interested in retrieving this way, if present, from the location bar using `$location.replace()`. This happens early enough that it doesn’t seem to cause any visual glitches or extra network requests.

File: kifi-backend/modules/shoebox/angular/src/common/originTracking.js [hunk 2]
```diff
@@ -3,8 +3,14 @@
...
 .factory('originTrackingService', [
-  function () {
+  'initParams',
+  function (initParams) {
     var pageOrigin = '';
+    var pageOrigins = {
+      xmp: 'extension/messagePane',
+      xsr: 'extension/searchResult',
+      xst: 'extension/socialTooltip'
+    };
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/common/originTracking.js:14]: This mechanism should be easy to reuse for any other entry points to our app by adding more mappings here.

**Comment by yipingliao** [comment]
:+1:

**Pull request merged**